lib/puppet/indirector/facts/facter.rb

```diff
@@ -21,6 +21,13 @@ def save(facts)
   # Lookup a host's facts up in Facter.
   def find(request)
     Facter.reset
+
+    # Add the puppetversion fact; this is done before generating the hash so it is
+    # accessible to custom facts.
+    Facter.add(:puppetversion) do
+      setcode { Puppet.version.to_s }
+    end
+
```

Comment:
The need for `puppetversion` comes from a custom fact in `stdlib`, so we definitely need to do it before `Facter.to_hash`. I chose this implementation route to preserve the previous behavior, where the local facts don't show up for custom facts, and the `puppetversion` fact doesn't show up for network devices.

Agreed that we should update the docs somehow; I assumed `puppetversion` was already in docs somewhere and usage hasn't really changed.